.github/workflows/ci.yml

```diff
@@ -8,64 +8,58 @@ jobs:
-     # run: docker run -e TRELLO_BOARD_ID=${{ secrets.TRELLO_BOARD_ID }} -e TRELLO_TOKEN=${{ secrets.TRELLO_TOKEN }} -e TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }} -e ID_LIST_NEWLIST=${{ secrets.ID_LIST_NEWLIST }} -e ID_LIST_TODO=${{ secrets.ID_LIST_TODO }} -e ID_LIST_DONE=${{ secrets.ID_LIST_DONE }} -e ID_LIST_DOING=${{ secrets.ID_LIST_DOING }} -e CARDS_TRELLODONE_CARD=${{ secrets.CARDS_TRELLODONE_CARD }} -e CARDS_TODO_CARD_TWO=${{ secrets.CARDS_TODO_CARD_TWO }} -e CARDS_TODO_CARD_THREE=${{ secrets.CARDS_TODO_CARD_THREE }} -e CARDS_TODO_CARD_ONE=${{ secrets.CARDS_TODO_CARD_ONE }} -e CARDS_DEVOPSTRELLOBOARD=${{ secrets.CARDS_DEVOPSTRELLOBOARD }} -e FLASK_APP=${{ secrets.FLASK_APP }} -e FLASK_ENV=${{ secrets.FLASK_ENV }} -e SECRET_KEY=${{ secrets.SECRET_KEY }}  -p 5100:5000 --mount type=bind,source="$(pwd)"/todo_app,target=/app/todo_app  todo-app:test /app/todo_app/SeleniumTest
-
       run: docker run -e FLASK_APP=${{ secrets.FLASK_APP }} -e FLASK_ENV=${{ secrets.FLASK_ENV }} -e SECRET_KEY=${{ secrets.SECRET_KEY }} -e TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }} -e TRELLO_TOKEN=${{ secrets.TRELLO_TOKEN }} -e TRELLO_BOARD_ID=${{ secrets.TRELLO_BOARD_ID }} -e ID_LIST_TODO=${{ secrets.ID_LIST_TODO }} -e ID_LIST_DONE=${{ secrets.ID_LIST_DONE }} -e ID_LIST_DOING=${{ secrets.ID_LIST_DOING }} -e CARDS_TRELLODONE_CARD=${{ secrets.CARDS_TRELLODONE_CARD }} -p 5100:5000 --mount type=bind,source="$(pwd)"/todo_app,target=/app/todo_app  todo-app:test /app/todo_app/SeleniumTest
 
+    - name: Build and push to heroku registry
+      run: heroku container:push web --app herukudevopsapp
 
     -  name: Build and push
        id: docker_build
        uses: docker/build-push-action@v2
        with:
           push: true
           tags: arpithadockerhub/app:latest
-         # tags: arpithadockerhub/dotnettemplate:latest
-
-# - uses: akhileshns/heroku-deploy@v3.12.12 # This is the action
-    #   with:
-    #       heroku_api_key: ${{secrets.HEROKU_API_KEY}}
-    #       heroku_app_name: "herukudevopsapp" #Must be unique in Heroku
-    #       heroku_email: ${{secrets.HEROKU_EMAIL}} 
-    #       justlogin: true
-
-    # - name: "Login to heroku registry"
-    #   run: heroku container:login
-
-    # - name: "Build and push to heroku registry"
-    #   run: heroku container:push web --app herukudevopsapp
+         # tags: arpithadockerhub/app:latest
+
+    - uses: akhileshns/heroku-deploy@v3.12.12    #  This is the action
+      with:
+          heroku_api_key: ${{secrets.HEROKU_API_KEY}}
+          heroku_app_name: "herukudevopsapp"
+          #Must be unique in Heroku
+          heroku_email: ${{secrets.HEROKU_EMAIL}} 
+          justlogin: true
+     
+    - name: "Login to heroku registry"
+      run: heroku container:login
 
-    # - name: "Releases the container"
-    #   run: heroku container:release web --app herukudevopsapp
+    - name: "Build and push to heroku registry"
+      run: heroku container:push web --app herukudevopsapp
```

Comment:
This will build the last target in your Dockerfile by default, which is your test one. You can specify the target if you push using the Docker CLI:

```bash
heroku container:login # if you haven't already
docker build --target production --tag registry.heroku.com/herukudevopsapp/web .
docker push registry.heroku.com/herukudevopsapp/web
```